airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -704,3 +705,122 @@ def lint_config(args) -> None:
+
+    This command scans the current configuration file for parameters that have been renamed,
+    removed, or had their default values changed in Airflow 3.0, and automatically updates
+    the configuration file.
+
+    CLI Arguments:
+        --section: str (optional)
+            Comma-separated list of configuration sections to update.
+            Example: --section core,database
+
+        --option: str (optional)
+            Comma-separated list of configuration options to update.
+            Example: --option sql_alchemy_conn,dag_concurrency
+
+        --ignore-section: str (optional)
+            Comma-separated list of configuration sections to ignore during update.
+            Example: --ignore-section webserver
+
+        --ignore-option: str (optional)
+            Comma-separated list of configuration options to ignore during update.
+            Example: --ignore-option check_slas
+
+        --verbose: flag (optional)
+            Enable detailed output, including a summary of changes and any ignored items.
+            Example: --verbose
+
+    Examples:
+        1. Update the entire configuration file:
+            airflow config update
+
+        2. Update only the 'core' and 'database' sections:
+            airflow config update --section core,database
+
+        3. Update only specific options:
+            airflow config update --option sql_alchemy_conn,dag_concurrency
+
+        4. Ignore updates for a specific section:
+            airflow config update --ignore-section webserver
+
+        5. Enable verbose output for more detailed information:
+            airflow config update --verbose
+
+    :param args: The CLI arguments for updating configuration.
+    """
+    console = AirflowConsole()
+    changes_applied = []
+
+    update_sections = args.section if args.section else None
+    update_options = args.option if args.option else None
+    ignore_sections = args.ignore_section if args.ignore_section else []
+    ignore_options = args.ignore_option if args.ignore_option else []
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
+            continue
+        if conf_section in ignore_sections or conf_option in ignore_options:
+            continue
+
+        if not conf.has_option(conf_section, conf_option, lookup_from_deprecated=False):
+            continue
+
+        current_value = conf.get(conf_section, conf_option)
+
+        if change.default_change and (str(current_value) != str(change.new_default)):
+            conf.set(conf_section, conf_option, str(change.new_default))
+            changes_applied.append(
+                f"Updated default value of '{conf_section}/{conf_option}' from '{current_value}' to '{change.new_default}'."
+            )
```

Comment:
We really want to only do this if the config is set in airflow.cfg. If it's from env vars or secret backend, we do not want to set it - it wouldn't be used anyways, and will just lead to confusion.

Same with the blocks below too.

Reply:
Ah, that isn't a bad idea at all!

Reply:
I have now changed the logic to change the configs in `airflow.cfg` only.

Reply:
I can create a separate PR for `airflow config lint`.